geode-pulse/src/main/java/org/apache/geode/tools/pulse/internal/PulseAppListener.java

```diff
@@ -44,18 +44,37 @@
 // @WebListener
 public class PulseAppListener implements ServletContextListener {
   private static final Logger logger = LogManager.getLogger();
-  private final ResourceBundle resourceBundle = Repository.get().getResourceBundle();
   private static final String GEODE_SSLCONFIG_SERVLET_CONTEXT_PARAM = "org.apache.geode.sslConfig";
 
-  private Properties pulseProperties;
-  private Properties pulseSecurityProperties;
+  private final boolean sysIsEmbedded;
+  private final Repository repository;
+  private final Function<String, Properties> functionLoadProperties;
+  private final ResourceBundle resourceBundle;
+
+  public PulseAppListener() {
+    this(Boolean.getBoolean(PulseConstants.SYSTEM_PROPERTY_PULSE_EMBEDDED), Repository.get(),
+        null);
```

Comment:
Is there a way to pass `loadProperties` here instead of `null`? (I'm guessing not, given that `loadProperties()` is an instance method and can't be made static.)

Reply:
I'd like to look a little closer also, I'm not super thrilled with the logic in the other constructor checking null.  Constructors with logic I think are not best practice (but if this was the best method then 🤷‍♀️)

Reply:
@demery-pivotal The answer is, no, we can't pass an instance method as an argument here because the object is not fully constructed at this point. I went with the approach of making the function static so we can pass it as an argument, which required adding ResourceBundle as a parameter.

Another approach would have been creating an interface e.g. IPropertiesLoader and passing that as a parameter in the overloaded constructor instead of the function. In this case however, that approach seems more complex because it requires creating an interface and two implementations of that interface.